mlir/lib/Support/Timing.cpp

```diff
@@ -125,40 +126,127 @@ struct TimeRecord {
     return *this;
   }
 
-  /// Print the current time record to 'os', with a breakdown showing
-  /// contributions to the give 'total' time record.
-  void print(raw_ostream &os, const TimeRecord &total) {
+  double wall, user;
+};
+
+class OutputStrategy {
```

Comment:
Nit: wrap the class in an anonymous namespace

Reply:
Actually it's already in such a namespace, I hadn't extended the GitHub diff scope...